lib/hooks/core/hook-http.js

```diff
@@ -35,6 +35,7 @@ function requestWrap(request) {
 
     var localOptions = (typeof options === 'string') ?
       url.parse(options) : options;
+    localOptions.headers = localOptions.headers || {};
```

Comment:
Lastly, if we are to do any object copying, this needs to happen *after* the root span checks below. i.e. only traces that are actually being sampled.